### PR TITLE
Add function to create oscar colassal dataset

### PR DESCRIPTION
**Pull request by kriangkraitan**
## Why this PR
Add function create oscar colassal dataset to openthaigpt-pretraining\src\data\scripts\internet\main.py
## Changes
- Write some changes here

## Related Issues
Close #

## Checklist
- [ ] PR should be in the [Naming convention](../../docs/PR_NAMING.md)
- [ ] Assign yourself in to Assigneees
- [ ] Tag related issues
- [ ] Constants name should be ALL_CAPITAL, function name should be snake_case, and class name should be CamelCase
- [ ] complex function/algorithm should have [Docstring](https://peps.python.org/pep-0257/)
- [ ] 1 PR should not have more than 200 lines changes (Exception for test files). If more than that please open multiple PRs
- [ ] At least PR reviewer must come from the task's team (model, eval, data)

### CODE REVIEW

File: src/data/scripts/internet/main.py [hunk 1]
```diff
@@ -148,29 +151,61 @@ def process_chunk_data(chunk):
...
+                try:
+                    with zstd.open(open(file_path, "rb"), "rt", encoding="utf-8") as f:
+                        id = 0  # Initialize ID for each file
+                        for row in f:
+                            item = json.loads(row)
+                            yield {
+                                "text": item["content"],
+                                "create_date": item["warc_headers"]["warc-date"],
+                                "source": "oscar_colossal_{}".format(
+                                    root.split("/")[-2]
```
Comment (new5558) [inline src/data/scripts/internet/main.py:191]: Can you name this line as a variable

File: src/data/scripts/internet/main.py [hunk 2]
```diff
@@ -148,29 +151,61 @@ def process_chunk_data(chunk):
...
+        for filename in files:
+            if filename.endswith(".jsonl.zst"):
+                file_path = os.path.join(root, filename)
+                try:
+                    with zstd.open(open(file_path, "rb"), "rt", encoding="utf-8") as f:
+                        id = 0  # Initialize ID for each file
+                        for row in f:
+                            item = json.loads(row)
+                            yield {
+                                "text": item["content"],
```
Comment (new5558) [inline src/data/scripts/internet/main.py:188]: Is this missing some keys?
image

Comment (kriangkraitan) [reply src/data/scripts/internet/main.py:189]: updated_date will create in process_chunk_data(chunk) line 87

File: src/data/scripts/internet/main.py (hunk 1 above)
Comment (kriangkraitan) [reply src/data/scripts/internet/main.py]: done

**Review by Chawak — APPROVED** [review]

**Review by new5558 — APPROVED** [review]

**Pull request merged**